src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -209,7 +210,10 @@ public URLClassPath(URL[] urls, @SuppressWarnings("removal") AccessControlContex
 
         this.unopenedUrls = unopenedUrls;
         this.path = path;
-        this.jarHandler = null;
+        // this URLClassPath constructor is solely for the app classloader, for which we use
+        // the system provided jar protocol handler implementation for loading jars
+        // in the classpath
```

Comment:
The apiNote on the constructor makes it clear that the constructor is for the application class path.  Also just to say that the term used in the URL docs is "built-in protocol handler". Combined then it might be clearer if the comment says that the application class loader must use the built-in protocol handler.